addon/internal-session.js

```diff
@@ -17,25 +17,24 @@ export default Ember.ObjectProxy.extend(Ember.Evented, {
     Ember.assert(`Session#authenticate requires the authenticator to be specified, was "${authenticator}"!`, !Ember.isEmpty(authenticator));
     let theAuthenticator = getOwner(this).lookup(authenticator);
     Ember.assert(`No authenticator for factory "${authenticator}" could be found!`, !Ember.isNone(theAuthenticator));
-    return new Ember.RSVP.Promise((resolve, reject) => {
+    return new RSVP.Promise((resolve, reject) => {
       theAuthenticator.authenticate.apply(theAuthenticator, args).then((content) => {
-        this._setup(authenticator, content, true);
-        resolve();
+        this._setup(authenticator, content, true).then(resolve, reject);
       }, (error) => {
-        this._clear();
-        reject(error);
+        this._clear().finally(() => {
+          reject(error);
+        });
```

Comment:
Does `finally` actually handle the rejected promise so that id doesn't appear in Ember Inspector's list of rejected promises and would also not break tests?